cpp/src/arrow/ipc/generate_tensor_fuzz_corpus.cc

```diff
@@ -41,7 +41,7 @@ using ::arrow::internal::PlatformFilename;
 Result<PlatformFilename> PrepareDirectory(const std::string& dir) {
   ARROW_ASSIGN_OR_RAISE(auto dir_fn, PlatformFilename::FromString(dir));
   RETURN_NOT_OK(::arrow::internal::CreateDir(dir_fn));
-  return std::move(dir_fn);
```

Comment:
I'm afraid this would meet same problem ( https://github.com/apache/arrow/pull/41025 ) in gcc 7.5.0, sigh...

Reply:
I don't think so. #41025 involved an implicit conversion from `unique_ptr` to `shared_ptr`, IIUC, which isn't the case here.

Reply:
So this would not casting implicit cast `PlatformFilename` to `Result<PlatformFilename>` or did I miss something?

Reply:
In #41025 the cast is from `std::unique_ptr` to `Result<std::shared_ptr>`, which is more complex (there are two cascaded casts).

Reply:
Let's try anyway.